spring-integration-core/src/main/java/org/springframework/integration/dsl/StandardIntegrationFlow.java

```diff
@@ -69,12 +70,35 @@
 
 	private final List<SmartLifecycle> lifecycles = new LinkedList<>();
 
+	private MessageChannel inputChannel;
+
 	private boolean running;
 
 	StandardIntegrationFlow(Map<Object, String> integrationComponents) {
 		this.integrationComponents = new LinkedHashMap<>(integrationComponents);
 	}
 
+	@Override
+	public void configure(IntegrationFlowDefinition<?> flow) {
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public MessageChannel getInputChannel() {
+		if (this.inputChannel == null) {
+			this.inputChannel =
+					this.integrationComponents.keySet()
+							.stream()
+							.filter(MessageChannel.class::isInstance)
+							.map(MessageChannel.class::cast)
+							.findFirst()
+							.orElseThrow(() -> new IllegalStateException("The 'IntegrationFlow' [" + this + "] " +
+									"doesn't start with 'MessageChannel' for direct message sending."));
```

Comment:
Is it really valid to search for the first channel in a flow? Shouldn't we simply reject subflows that don't start with a channel?

Reply:
I thought about that one more time, just because the previous version really rejected flows from non-channel in the beginning. But right now I think it doesn't matter: we always are good to send a message to the output channel of any message producer in the beginning of the flow. In other words essentially we are selecting here the *first* channel anyway.
See this:
```
IntegrationFlows.from(MessageSource).handle().get();
...
IntegrationFlows.from(MessageProducer).handle().get();
```
So, what we select here is an `inputChannel` of that `handle()`. What is the point to reject it?

That's might not be obvious at a glance, but it becomes reasonable to reuse existing flows even if they are not starting from the channel. We may document it though or just leave as silent feature until someone will ask :popcorn:

Reply:
OK; makes sense; did you see the comment in my review overview?